src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -435,6 +436,105 @@ public void createSnapshot(Snapshot snapshot, List<String> rowIds) throws Interr
         snapshotViewCreation(datasetBqDatasetName, snapshotName, snapshot, projectId, bigQuery, bigQueryProject);
     }
 
+    private static final String insertAllLiveViewDataTemplate =
+        "INSERT INTO `<project>.<snapshot>.<dataRepoTable>` " +
+            "(<dataRepoTableId>, <dataRepoRowId>) <liveViewTables>";
+
+    private static final String getLiveViewTableTemplate =
+        // TODO pull insert out and loop thru rest w UNION ()
+            "(SELECT '<tableId>', <dataRepoRowId> FROM `<project>.<dataset>.<liveView>`)";
+
+    private static final String mergeLiveViewTablesTemplate =
+        "<selectStatement1> UNION ALL <selectStatement2>";
+
+    private static final String validateSnapshotSizeTemplate =
+        "SELECT * FROM <project>.<snapshot>.<dataRepoTable>";
```

Comment:
Looks like this is only used to check that the snapshot is non-empty, could it be `SELECT COUNT(1)` instead to avoid pointless scanning and reduce the result size?